db/snapshot/backup_task.cc

```diff
@@ -97,7 +98,12 @@ future<> backup_task_impl::do_backup() {
         // Parallelism is implicitly controlled in two ways:
         //  - s3::client::claim_memory semaphore
         //  - http::client::max_connections limitation
-        std::ignore = _client->upload_file(component_name, destination, _progress, &_as).handle_exception([comp = component_name, &ex] (std::exception_ptr e) {
+        std::ignore = _client->upload_file(component_name, destination, _progress, &_as).then([component_name] {
+            // Delete the uploaded component to:
```

Comment:
It should happen conditionally based on also-to-be-added API parameter

Reply:
> named "remove_on_uploaded"

A suggestion -- it can be named "move_files" or something as this is what it effectively going to do

Reply:
I am not familiar with such an option today.
To the best of my knowledge rclone `move` files - there is no option to ask it to `copy`.

What is the use case you have in mind?
Keeping the snapshot after uploading it means that now we have two copies of the same SSTables in two places. We'll need to decide which one to use in various scenarios. Why?